### PR TITLE
[Core] Check cmake version in stub generation

### LINKED ISSUE #10712: Errors when configuring
**Description**
Hello,

I tried to clone and compile Kratos on my Ubuntu on Windows, and btw my default python is 3.8 version. but when running the bash script I get the following error and I don't know what to do and what is the issue. many thanks.

**Scope**
-- clipper : /home/.../Kratos/external_libraries/clipper
-- tinyexpr : /home/.../Kratos/external_libraries/tinyexpr/tinyexpr/
python libraries to be installed are: /usr/lib/x86_64-linux-gnu/libpython3.8.so.1.0
CMake Error at scripts/post_install/CMakeLists.txt:5 (cmake_path):
  Unknown CMake command "cmake_path".

-- Configuring incomplete, errors occurred!
See also "/home/.../Kratos/build/Release/CMakeFiles/CMakeOutput.log".
make: *** No rule to make target 'install'.  Stop.

### PR DESCRIPTION
**Pull request by sunethwarna**
**📝 Description**
This PR makes the installation to ignore stub generation if `CMake` version is less than 3.20.0. I found a small bug in the stub generation python file which is rectified in this PR.

closes #10712 

**🆕 Changelog**
- Check the cmake version for stub generation
- Bug fix for python stub generation

### CODE REVIEW

File: scripts/post_install/stub_generation.py [hunk 1]
```diff
@@ -42,7 +42,7 @@ def __FindCPPModuleImportsInPythonModules(current_path: Path, cpp_module_names_l
 
                 data = data.replace(" ", "")
                 for i, cpp_module_name in enumerate(cpp_module_names_list):
-                    if "\nfrom{:s}import*".format(cpp_module_name) in data:
+                    if "\nfrom{:s}import*".format(cpp_module_name) in data or "from{:s}import*".format(cpp_module_name) in data:
```
Comment (matekelemen) [inline scripts/post_install/stub_generation.py]: you must have known that I was going to suggest regex :laughing:

You could try something like [this](https://regex101.com/r/at6bOp/1) for example:
```suggestion
for match in re.finditer(R"from +(\w+) +import +\*", data):
    cpp_module_name = match.group(1)
    if cpp_module_name in cpp_module_names_list:
```

*don't commit the suggestion directly through github; the indenting is all messed up*

**Comment by sunethwarna** [comment]
@matekelemen This is ready :D

File: scripts/post_install/stub_generation.py [hunk 2]
```diff
@@ -200,8 +217,11 @@ def Main():
             shutil.rmtree(str(kratos_library_path / "KratosMultiphysics"))
 
 if __name__ == "__main__":
+    error_and_warning_outut_file = f"{sys.argv[1]}/stub_generation_errors_and_warnings.txt"
     if "--quiet" in sys.argv: # suppress output from Kratos imports
         args = [arg for arg in sys.argv if arg != "--quiet"]
-        subprocess.run([sys.executable] + args, stdout = subprocess.PIPE, stderr = subprocess.PIPE)
+        output = subprocess.run([sys.executable] + args, stdout = subprocess.PIPE, stderr = subprocess.PIPE)
+        for matched_group in re.finditer(r"(Error|Warning|error|warning)(.*?\\n)", str(output.stdout) + str(output.stderr)):
```
Comment (matekelemen) [inline scripts/post_install/stub_generation.py:224]:
```suggestion
        for matched_group in re.finditer(r"([eE]rror|[wW]arning)(.*?\\n)", str(output.stdout) + str(output.stderr)):

```

**Review by matekelemen — APPROVED** [review]
just fyi, no need to commit

**Pull request merged**